core/commands/storage.go

```diff
@@ -132,22 +154,44 @@ Receive proofs as collateral evidence after selected nodes agree to store the fi
 		lf, _ := req.Options[leafHashOptionName].(bool)
 		if lf == false {
 			if len(req.Arguments) != 1 {
-				return fmt.Errorf("need one and only one root hash")
+				return fmt.Errorf("need one and only one root file hash")
 			}
 			// get leaf hashes
 			rootHash = req.Arguments[0]
 			hashToCid, err := cidlib.Parse(rootHash)
 			if err != nil {
 				return err
 			}
-			rp, err := api.ResolvePath(req.Context, path.IpfsPath(hashToCid))
+			rootPath := path.IpfsPath(hashToCid)
+			// check to see if a replicated file using reed-solomon
+			mbytes, err := api.Unixfs().GetMetadata(req.Context, rootPath)
+			if err != nil {
+				return fmt.Errorf("file must be reed-solomon encoded: %s", err.Error())
+			}
+			var rsMeta chunker.RsMetaMap
+			err = json.Unmarshal(mbytes, &rsMeta)
+			if err != nil {
+				return fmt.Errorf("file must be reed-solomon encoded: %s", err.Error())
+			}
+			if rsMeta.NumData == 0 || rsMeta.NumParity == 0 || rsMeta.FileSize == 0 {
+				return fmt.Errorf("file must be reed-solomon encoded: metadata not valid")
+			}
+			// use unixfs layer helper to grab the raw leaves under the data root node
+			// higher level helpers resolve the enrtire DAG instead of resolving the root
+			// node as it is required here
+			rn, err := api.ResolveNode(req.Context, rootPath)
 			if err != nil {
 				return err
 			}
-			cids, err := api.Object().Links(req.Context, rp)
+			nodes, err := unixfs.GetChildrenForDagWithMeta(req.Context, rn, api.Dag())
 			if err != nil {
 				return err
 			}
+			// The second child is always data root node
+			cids := nodes[1].Links()
```

Comment:
are/will we repeat this somewhere else? y --> maybe add a helper func to get the data root node to wrap up the magic number here

Reply:
It used to be only for unixfs-related functions so the magic is contained in that module.... but I have to use it here to get the raw node (not all the resolved path).

@steveyeom are you ok if I modify `GetChildrenForDagWithMeta` to return a tuple of `(ipld.Node, ipld.Node, error)` instead of a 2-length slice? Because it's hard to infer the nodes[1] from the caller.